src/test/java/test/tesng1046/TestClassSample.java

```diff
@@ -0,0 +1,52 @@
+package test.tesng1046;
+
+import org.testng.IAlterTestName;
+import org.testng.IHookCallBack;
+import org.testng.IHookable;
+import org.testng.ITestResult;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertTrue;
+
+public class TestClassSample implements IHookable {
+
+    @DataProvider (name = "dp", parallel = true)
+    public Object[][] getTestData() {
+        return new Object[][] {
+            {1}, {2}, {3}, {4}, {5}
+        };
+    }
+
+    @Test (dataProvider = "dp")
+    public void testSample1(int num) {
+        assertTrue(num != 0);
+    }
+
+    @Test (dataProvider = "dp")
+    public void testSample2(int num) {
+        assertTrue(num != 0);
```

Comment:
I don't know what is @cbeust recommendation about that but I prefer `assertEquals` or `assertNotEquals` because the message won't hide some information (the passed value of `num`).
I think testng tests should be the place where testng best practices (again, @cbeust should give us his opinion) are used